### PR TITLE
task/send_image/take_image

### PR DESCRIPTION
**Pull request by teamagam1**

### CODE REVIEW

**Comment by review-ninja** [comment]
<a href="https://app.review.ninja/teamagam/GimelGimel/pull/23" target="_blank">ReviewNinja

File: app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        File tempDir = new File(context.getExternalCacheDir(), TEMP_DIR);
+        if(!tempDir.exists()) {
+            tempDir.mkdirs();
+        }
+
+        String timeStamp = new SimpleDateFormat(DATE_FORMAT).format(new Date());
+        String name = IMAGE_TEMP_PREFIX + timeStamp;
+
+        File tempFile = File.createTempFile(name, IMAGE_TEMP_TYPE, tempDir);
+        tempFile.delete();
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java]: This seems fishy.
Do you really have to create and delete a file to get a temp URI ?

File: app/src/main/java/com/teamagam/gimelgimel/app/network/services/IImageSender.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import android.net.Uri;
+
+import com.teamagam.gimelgimel.app.view.viewer.data.geometries.PointGeometry;
+
+/**
+ * Created on 5/22/2016.
+ * Interface for sending and uploading images
+ */
+public interface IImageSender{
+    void sendImage(Uri imageUri, PointGeometry location, long imageTime);
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/services/IImageSender.java]: Is this interface for uploading an image or sending an ImageMessage?

Either way, renaming and documentation is needed !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 3]
```diff
@@ -60,11 +70,14 @@
...
     private BroadcastReceiver mLocationReceiver;
 
+    private Uri mImageUri;
+
     @BindView(R.id.camera_fab)
     FloatingActionButton mCameraFab;
 
     @BindView(R.id.message_fab)
     FloatingActionButton mMessageFab;
+    private IImageSender imageSender;
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: Data members naming conventions: mCamelCase !fix

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java [hunk 4]
```diff
@@ -113,14 +126,47 @@ public void onReceive(Context context, Intent intent) {
...
     }
 
+    @OnClick(R.id.message_fab)
+    public void sendMessage() {
+        new SendMessageDialogFragment().show(getFragmentManager(), "sendMessageDialog");
+    }
+
     @OnClick(R.id.camera_fab)
     public void takePicture() {
         Toast.makeText(mApp, "Take Picture", Toast.LENGTH_SHORT).show();
```
Comment (BarAgam) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: Seems like this toast is redundant now

File: app/src/main/java/com/teamagam/gimelgimel/app/network/services/IImageSender.java (hunk 2 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/network/services/IImageSender.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java (hunk 1 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java]: no, it is to be sure that we can write there.

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 3 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !fixed

File: app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java (hunk 4 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/view/fragments/ViewerFragment.java]: !completed

File: app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java (hunk 1 above)
Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java]: !resolved

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java]: !fix

Comment (teamagam1) [inline app/src/main/java/com/teamagam/gimelgimel/app/utils/ImageUtil.java]: !fixed

**Pull request merged**